packages/typescript/checked_in_ts_project.bzl

```diff
@@ -31,19 +37,43 @@ def checked_in_ts_project(name, src, checked_in_js = None, **kwargs):
         **kwargs
     )
 
-    # Don't trigger clang-format on the output js
-    # Make sure we don't add any lines though, since that would
-    # break the sourcemap
-    native.genrule(
-        name = "_%s_skip_formatting" % name,
-        srcs = [name],
-        outs = ["_%s_no_format.js" % name],
-        cmd = """echo -n "/* THIS FILE GENERATED FROM .ts; see BUILD.bazel */ /* clang-format off */" > $@; cat $< >> $@""",
+    append_skip_formatting_header(
+        name = "%s_js_out" % name,
+        input = name,
     )
 
     # Assert that we kept the index.js up-to-date when changing the TS code
     generated_file_test(
         name = "%s_check_compiled" % name,
-        generated = "_%s_no_format.js" % name,
+        generated = "%s_js_out" % name,
         src = checked_in_js,
     )
+
+    if checked_in_types != None:
+        native.filegroup(
+            name = "%s_types" % name,
+            output_group = "types",
+            srcs = [name],
+        )
+
+        append_skip_formatting_header(
+            name = "%s_types_out" % name,
+            input = "%s_types" % name,
+        )
+
+        generated_file_test(
+            name = "%s_check_types" % name,
+            generated = "%s_types_out" % name,
+            src = checked_in_types,
+        )
+
+def append_skip_formatting_header(name, input):
```

Comment:
love this refactoring. You clearly learned things from Jeremy Elbourn :)

maybe some bits of this could land as a prefactoring to make the review easier?